# **Skill Assessment Quiz**

### Question No.01

In [ ]:
import sympy
import math
from sympy import solve, Eq, symbols,I
from sympy.abc import s,alpha
import numpy as np
import cmath

print(solve(Eq(1/(s**2+1.4**2),(I*alpha/(s+1.4*I))-(I*np.conj(alpha)/(s-1.4*I))),alpha))



[0.357142857142857]


### Question No.02

In [ ]:
import numpy as np

eq=[1,2*2*1.1,1.1**2]

roots_all= np.roots(eq)
print(roots_all)

print("The largest root is ",max(roots_all))

[-4.10525589 -0.29474411]
The largest root is  -0.294744111674235


### Question No.03

In [7]:
import sympy
import math
from sympy import *
from sympy.abc import s,alpha,p
import numpy as np
import cmath

x = Eq(1/(s**2+2*0.6*1.5*s+1.5**2),(I*alpha/(s+p))-(I*np.conj(alpha)/(s+np.conj(p))))
print(sympy.latex(x))
print()

leftSide= 1/(s**2+2*0.6*1.5*s+1.5**2)
partialFractions=apart(leftSide,full=True).doit()
print(sympy.latex(partialFractions))

\frac{1}{s^{2} + 1.8 s + 2.25} = \frac{i \alpha}{p + s} - \frac{i \overline{\alpha}}{s + \overline{p}}

\frac{1.0 \cdot \left(5.55111512312578 \cdot 10^{-17} + 0.416666666666667 i\right)}{s + 0.9 + 1.2 i} + \frac{1.0 \cdot \left(5.55111512312578 \cdot 10^{-17} - 0.416666666666667 i\right)}{s + 0.9 - 1.2 i}


$\frac{1}{s^{2} + 1.8 s + 2.25} = \frac{i \alpha}{p + s} - \frac{i \overline{\alpha}}{s + \overline{p}}$

$\frac{1.0 \cdot \left(5.55111512312578 \cdot 10^{-17} + 0.416666666666667 i\right)}{s + 0.9 + 1.2 i} + \frac{1.0 \cdot \left(5.55111512312578 \cdot 10^{-17} - 0.416666666666667 i\right)}{s + 0.9 - 1.2 i}$

### Question No.04

In [28]:
import sympy
import math
import numpy as np
import cmath

A=[0.2,1.7,-1.4]
B=[0,-0.2,-0.7]

vector1=np.cross(A,B)
print(np.linalg.norm(vector1))

1.4771932845772078


### Question No.05

In [12]:
import sympy as sp
from sympy import Matrix
from sympy import *
from sympy.abc import omega

x = sp.symbols('x', real=True,domain=(1,5))
m= Matrix([[0,1],[-omega**2,-2*x*omega]])

print(sympy.latex(m.eigenvals()))
print()
print(sympy.latex(m.eigenvects()))


\left\{ \omega \left(- x - \sqrt{\left(x - 1\right) \left(x + 1\right)}\right) : 1, \  \omega \left(- x + \sqrt{\left(x - 1\right) \left(x + 1\right)}\right) : 1\right\}

\left[ \left( \omega \left(- x - \sqrt{\left(x - 1\right) \left(x + 1\right)}\right), \  1, \  \left[ \left[\begin{matrix}- \frac{2 x}{\omega} - \frac{- x - \sqrt{\left(x - 1\right) \left(x + 1\right)}}{\omega}\\1\end{matrix}\right]\right]\right), \  \left( \omega \left(- x + \sqrt{\left(x - 1\right) \left(x + 1\right)}\right), \  1, \  \left[ \left[\begin{matrix}- \frac{2 x}{\omega} - \frac{- x + \sqrt{\left(x - 1\right) \left(x + 1\right)}}{\omega}\\1\end{matrix}\right]\right]\right)\right]


$\left\{ \omega \left(- x - \sqrt{\left(x - 1\right) \left(x + 1\right)}\right) : 1, \  \omega \left(- x + \sqrt{\left(x - 1\right) \left(x + 1\right)}\right) : 1\right\}$

$\left[ \left( \omega \left(- x - \sqrt{\left(x - 1\right) \left(x + 1\right)}\right), \  1, \  \left[ \left[\begin{matrix}- \frac{2 x}{\omega} - \frac{- x - \sqrt{\left(x - 1\right) \left(x + 1\right)}}{\omega}\\1\end{matrix}\right]\right]\right), \  \left( \omega \left(- x + \sqrt{\left(x - 1\right) \left(x + 1\right)}\right), \  1, \  \left[ \left[\begin{matrix}- \frac{2 x}{\omega} - \frac{- x + \sqrt{\left(x - 1\right) \left(x + 1\right)}}{\omega}\\1\end{matrix}\right]\right]\right)\right]$